app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -33,7 +41,11 @@ def command_headers
     }
 
     if registration_params['jwt_expiration'].present?
-      jwt_args[:expiration] = registration_params['jwt_expiration'].to_i.hours.to_i if registration_params['jwt_expiration'] != 'unlimited'
+      if is_token_valid
+        jwt_args[:expiration] = registration_params['jwt_expiration'].to_i.hours.to_i unless unlimited
+      else
+        raise ArgumentError, _("Error: Option --jwt-expiration: The value must be between 0 to 999999 hours. 0 means 'unlimited'.")
```

Comment:
Some nitpicks:
* The API has no option `--jwt-expiration`, that's a hammer thing. I think here we should talk about the `jwt_expiration` *parameter*.
* I think it's more correct to raise a `Foreman::Exception` like this: `raise ::Foreman::Exception.new(N_("the text"))` (see https://projects.theforeman.org/projects/foreman/wiki/Translating#Exceptions)
* Why 999999? This seems like an arbitrary number without any meaning? It translates to roughly 114 years which seems rather long. We either should have a "sensible" limit, like a year or two. Or no limit at all (my preference).

Reply:
Thank you so much @evgeni , I'll update the exception, Foreman exception is definitely more valid. 
 I used 999999 because that's the limit set from the [UI validations](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/routes/RegistrationCommands/RegistrationCommandsPage/components/fields/TokenLifeTime.js#L18) I think when are  conventionally accepting unlimited, then 999999 hours(which is still limited) might make sense. I could remove the limit all together but that would not incline with the UI.